### DISCUSSION TITLE
strange background color during jpg to png conversion

### DISCUSSION THREAD

**Discussion by cbuxbaum** [ImageMagick/ImageMagick#4833] [Help]
Hello, Can anyone tell me what is happening with this conversion?  I have attached the original jpg and the png result.  The command I used is a simnple jog to png convert.  Here is the version info:

Version: ImageMagick 7.1.0-20 Q16-HDRI x86_64 2022-01-22 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
Compiler: gcc (4.2)

Installed on a Mac.

and the command line:

convert WN21-KT-531CTQR-01.jpg png:test.png

![WN21-KT-531CTQR-01](https://user-images.githubusercontent.com/4314907/153607493-023c3cf8-5e4b-4cc7-9b8b-17ff373fd180.jpg)

![test](https://user-images.githubusercontent.com/4314907/153607724-fc1fb8f3-6f2b-4b53-948b-7aa657c36d4c.png)

Additionally, I believe this image cannot be handled by java ImageIO (CMYK bug), but when handled by twelvemonkeys, the colors are more or less the same (no bad background) but not completely accurate.

Thanks for any help!

**Comment by snibgo** [comment]
You have v7 IM, but you use the v6 command "convert". Why? If there is no good reasons, I suggest you use the v7 command "magick" instead.

Did you build IM yourself? If so, are you using current versions of the libraries, especially PNG and JPEG?

Your input JPEG encodes pixels as CMYK. PNG cannot record CMYK, so IM automatically converts the image to sRGB. As a workaround, you might try an explicit conversion:
```
magick WN21-KT-531CTQR-01.jpg -colorspace sRGB png:test.png
```

**Reply by cbuxbaum** [reply]
lrwxr-xr-x  1 cbuxbaum  admin      6 Jan 22 09:32 convert -> magick
I believe I installed this version with brew or downloaded from IM.  However, we get similar results on other platforms.

Tried the colorspace arg, and get the same result.  Thanks for your suggestions.

**Comment by fmw42** [comment]
I tried your command on IM 6.9.12-38 and IM 7.1.0-23 with "convert" on Mac OSX Sierra and both output a correct result. I agree with user snibgo, that it would be better to explicitly convert from CMYK to sRGB. You could also convert with profiles. See if either profiles or -colorspace sRGB helps. If not, then perhaps upgrade your jpeg and png delegates and/or upgrade ImageMagick. Also if using IM 7,, then use "magick" not "convert" unless you have a symbolic link between the two.

**Comment by snibgo** [comment]
Sorry, I can't reproduce the problem using IM v7.1.0-20 or earlier versions.

Something is seriously wrong with your IM. I suggest testing simpler commands such as "magick rose: x.png", and creating a CMYK file and seeing if you can convert it back.
